apps/frontend/src/database/hooks/useDatabase.tsx

```diff
@@ -0,0 +1,68 @@
+import useSessionState from "@/modules/users/hooks/useSessionState";
+import useUserState from "@/modules/users/hooks/useUserState";
+import { createSelectors } from "@/utils/selectors";
+import { useCallback, useEffect, useState } from "react";
+import { DB, createDb, removeDb } from "..";
+import { useDbNamesStore } from "../states/dbNames";
+
+export default function useDatabase() {
+    const { user } = useUserState();
+    const { session } = useSessionState();
+    const dbNamesStore = createSelectors(useDbNamesStore);
+    const addDbName = dbNamesStore.use.add();
+    const clearDbNames = dbNamesStore.use.clear();
+    const dbNames = dbNamesStore.use.names();
+    const [db, setDb] = useState<DB | null>(null);
+
+    useEffect(() => {
+        const createDbForCurrentOrg = async () => {
+            if (!user || !session) return null;
+
+            const org = user.organizations?.edges.find(
+                ({ node }) => node.slug === session.organization.slug,
+            )?.node;
+            if (!org) return null;
+
+            const dbName = `integraflow-${org.slug}`;
+            const database = await createDb(dbName);
+
+            org.projects?.edges.forEach(async (edge) => {
+                await database.collections.projects.upsert({
+                    id: edge.node.id,
+                    name: edge.node.name,
+                    timezone: edge.node.timezone,
+                });
+            });
+
+            database.collections.viewer.upsert({
```

Comment:
@AfeezGL Why are persisting viewer in the DB? Do we use it anywhere? It's already in local storage, so I think that's enough.

Reply:
Okay.